src/ws/cockpitauth.c

```diff
@@ -1110,7 +1113,14 @@ cockpit_session_launch (CockpitAuth *self,
            g_str_equal (type, "tls-cert"))
     {
       if (command == NULL && unix_path == NULL)
-        command = cockpit_ws_session_program;
+        {
+          if (cockpit_ws_session_program)
+            command = cockpit_ws_session_program;
+          else if (g_file_test (WS_SESSION_SOCKET, G_FILE_TEST_EXISTS))
```

Comment:
Any reason you're checking this?

I'd prefer to use the socket unconditionally unless the user has otherwise configured a program to run for themselves.  Keep in mind: in our default install, `cockpit-session` isn't going to be setuid anymore, so it's going to be broken as a fallback (and will just make debugging things more difficult by showing a different error message than the actual root cause).

Reply:
Some tests start cockpit-ws directly and failed, [like this](https://cockpit-logs.us-east-1.linodeobjects.com/pull-16808-236785b9-20241107-134041-fedora-40-networking/log.html#54). The main problem is that the socket just won't exist if you don't use `cockpit.socket`, and at least currently you can't `systemctl enable cockpit-session.socket`. I don't really want to do that either.

Of course this would only work if you run cockpit-ws as root (the topmost commit message mentions that).

This needs some more thought..

Reply:
Reworked -- cockpitauth.c basically went back to your original version, and I adjusted the tests accordingly instead. I also added a new integration test to check "direct cockpit-session spawning" instead. It's not a use case which we want to document or encourage, but let's at least make sure it doesn't break.